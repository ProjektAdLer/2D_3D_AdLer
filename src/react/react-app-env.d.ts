/// <reference types="react-scripts" />

// Images
declare module '*.jpg';
declare module '*.png';
declare module '*.env';

// 3D types
declare module '*.glb';
declare module '*.stl';

declare module '*.jpeg';
declare module '*.gif';
declare module '*.mp4';
