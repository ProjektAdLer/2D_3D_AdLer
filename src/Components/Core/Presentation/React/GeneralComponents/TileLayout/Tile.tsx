import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";

type TileProps = {
  title?: string;
  image: string;
  onClick: () => void;
};

export default function Tile(props: TileProps) {
  return (
    <StyledButton
      className="flex flex-col"
      containerClassName="w-full h-full"
      shape="freeFloatCenter"
      onClick={props.onClick}
      title={props.title}
    >
      <div className="flex flex-col">
        <img
          className="w-full h-full"
          src={props.image}
          alt={props.title + "Thumbnail"}
        />
        {/* {props.title && <p onClick={props.onClick}>{props.title}</p>} */}
      </div>
    </StyledButton>
  );
}
