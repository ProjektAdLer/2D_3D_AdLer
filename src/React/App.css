body {
  margin: 0;
  height: 100vh;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

div{
  width: 100%;
  height: 100%;
}

section{
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 120px;
  height: 100vh;
  background-color: #fff;
}

button{
  width: 100px;
  margin: 10px;
  z-index: 5;
  position: fixed;
  left: 0;
  top: 0;
}

canvas {
  width: calc(100% - 100px);
  height: 100%;
  margin-left: 100px;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-content {
  width: 1000px;
  background-color: #fff;
}

.modal-header,
.modal-footer {
  padding: 10px;
  height: 60px;
}

.modal-title {
  margin: 0;
}

.modal-body {
  padding: 10px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}
