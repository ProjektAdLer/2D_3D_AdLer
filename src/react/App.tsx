import React from 'react';
import './App.css';
import BabylonCanvas from './components/BabylonCanvas';

function App() {
	return (
		<React.Fragment>
			<BabylonCanvas style={{ width: '100%', position: 'absolute' }} />
		</React.Fragment>
	);
}

export default App;
