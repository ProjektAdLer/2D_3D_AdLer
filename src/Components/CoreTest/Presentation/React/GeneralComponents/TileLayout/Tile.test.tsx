import { render } from "@testing-library/react";
import React from "react";
import Tile from "../../../../../Core/Presentation/React/GeneralComponents/TileLayout/Tile";

const img = require("../../../../../../Assets/avatarEditorThumbnails/hair/hairstyles/aa-hair-backhead.png");

describe("Tile", () => {
  test("should render", () => {
    const renderResult = render(<Tile image={img} onClick={() => {}} />);

    expect(renderResult.container).toMatchSnapshot();
  });

  test("click calls given onClick function", () => {
    const onClick = jest.fn();
    const renderResult = render(<Tile image={img} onClick={onClick} />);

    renderResult.getByRole("button").click();

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
