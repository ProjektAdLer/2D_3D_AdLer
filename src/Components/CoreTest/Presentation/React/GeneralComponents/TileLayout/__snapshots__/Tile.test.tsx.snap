// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile should render 1`] = `
<div>
  <div
    class=""
  >
    <button
      class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
    >
      <div
        class="flex flex-col"
      >
        <img
          alt="undefinedThumbnail"
          src="Hair_backhead_256.png"
        />
      </div>
    </button>
  </div>
</div>
`;
